### ISSUE TITLE
Fixed intermittent part of the large packet is lost

### ISSUE THREAD

**Issue by crazykooler** [joelcox/codeigniter-redis#29]
in __construct() function

``` php
$this->_connection = @fsockopen($this->_ci->config->item('redis_host'), $this->_ci->config->item('redis_port'), $errno, $errstr, 3);
```

to

``` php
$this->_connection = @stream_socket_client("tcp://".$this->_ci->config->item('redis_host').":".$this->_ci->config->item('redis_port'), $errno, $errstr, 3);
```

and in _bulk_reply() function

``` php
private function _bulk_reply()
{
    $value_length = (int) fgets($this->_connection);

    if($value_length > 0 && $this->_connection) {
        return fgets($this->_connection, $value_length + 1);
    }
}
```

**Comment by joelcox** [comment]
Hey Kim,

Thanks for opening this issue. Really curious if this helps with #17 at all.

Joël

**Comment by crazykooler** [comment]
Thanks your stuff :) :+1: 
I think there is a better way for me, I will help this
But, this solve everything related to the `stream_socket_client` created

**Comment by joelcox** [comment]
Hi Kim,

I'm really sorry that I haven't looked into this yet. I reserved some time later this week to look into your suggestions.

Joël

**Comment by joelcox** [comment]
I added your changes but this doesn't seem to fix #17.

https://github.com/joelcox/codeigniter-redis/tree/issue-29
https://travis-ci.org/joelcox/codeigniter-redis/jobs/5534631

**Comment by joelcox** [comment]
Hi Kim, care to pitch in on these changes?

**Comment by swznd** [comment]
sorry, I didn't read this first. it may be duplicate with my new pull request #34

**Comment by joelcox** [comment]
Fixed in #34.
